l1-contracts/deploy-scripts/PrepareZKChainRegistrationCalldata.s.sol

```diff
@@ -0,0 +1,286 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+// solhint-disable no-console
+
+import {Script, console2 as console} from "forge-std/Script.sol";
+import {stdToml} from "forge-std/StdToml.sol";
+
+import {IBridgehub} from "contracts/bridgehub/IBridgehub.sol";
+import {Bridgehub} from "contracts/bridgehub/Bridgehub.sol";
+import {L2ContractHelper} from "contracts/common/libraries/L2ContractHelper.sol";
+import {AddressAliasHelper} from "contracts/vendor/AddressAliasHelper.sol";
+import {L1SharedBridge} from "contracts/bridge/L1SharedBridge.sol";
+import {IGovernance} from "contracts/governance/IGovernance.sol";
+import {Utils} from "./Utils.sol";
+
+contract PrepareZKChainRegistrationCalldataScript is Script {
+    using stdToml for string;
+
+    address internal constant ADDRESS_ONE = 0x0000000000000000000000000000000000000001;
+
+    struct Config {
+        address chainAdmin;
+        address bridgehub;
+        address stateTransitionProxy;
+        uint256 chainId;
+        uint256 eraChainId;
+        uint256 bridgehubCreateNewChainSalt;
+        address baseToken;
+        bytes diamondCutData;
+        address governance;
+        address l1SharedBridgeProxy;
```

Comment:
are these still separate? (I thought we have a shared bridge). Anyways, IMHO should be fetched from bridgehub.

Reply:
it is just the same with erc20 bridge as with era_chain_id, not really relevant, but required for L2 bridge init

Reply:
from what I understand -- we used the erc20BridgeProxy as a backwards compatibility for era..

So - if you are deploying a non era chain - this is actually not used and can be set to 0 -- please lookg at the L2SharedBridge code.